### PR TITLE
NuttX Discovery profile [6001]

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request fixes the Discovery profile compilation over NuttX platforms.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -99,14 +99,16 @@ find_package(microcdr 1.1.0 EXACT REQUIRED PATHS ${PROJECT_BINARY_DIR}/temp_inst
...
+    set(UCLIENT_PLATFORM_POSIX ON)
     set(TRANSPORT_TYPES "UDP" "TCP" "SERIAL")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-    set(PLATFORM_NAME_WINDOWS ON)
+    set(UCLIENT_PLATFORM_WINDOWS ON)
     set(TRANSPORT_TYPES "UDP" "TCP")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Generic")
     if(PLATFORM_NAME STREQUAL "nuttx")
-        set(PLATFORM_NAME_NUTTX ON)
+        set(UCLIENT_NUTTX_PLATFORM ON)
```
Comment (BorjaOuterelo) [inline CMakeLists.txt]: Why is it not like other platforms?
UCLIENT_PLATFORM_*

File: include/uxr/client/profile/transport/udp/udp_transport_posix.h [hunk 2]
```diff
@@ -12,8 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifndef _UXR_CLIENT_UDP_TRANSPORT_LINUX_H_
-#define _UXR_CLIENT_UDP_TRANSPORT_LINUX_H_
+#ifndef UXR_CLIENT_UDP_TRANSPORT_LINUX_H_
```
Comment (BorjaOuterelo) [inline include/uxr/client/profile/transport/udp/udp_transport_posix.h]: Shouldn't it be UXR_CLIENT_UDP_TRANSPORT_POSIX_H_

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/8/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/6/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/49/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/50/)

**Pull request merged**